pytrip/vdx.py

```diff
@@ -204,12 +212,45 @@ def voi_names(self):
         names = [voi.name for voi in self.vois]
         return names
 
-    def add_voi(self, voi):
+    def add_voi(self, voi, added_manually=False):
         """ Appends a new voi to this class.
+        If added_manually flag is True, sets voi color as one from spare ones.
+        If there is no spare colors, calls assign_voi_colors
 
         :param Voi voi: the voi to be appended to this class.
+        :param bool added_manually: flag that changes way VOI is added, default False
         """
-        self.vois.append(voi)
+        if added_manually:
+            if len(self._spare_voi_colors) > 0:
+                color = self._spare_voi_colors.pop()
+                voi.set_color(color)
+                self.vois.append(voi)
+            else:
+                self.assign_voi_colors()
+        else:
+            self.vois.append(voi)
+
+    def assign_voi_colors(self, k=3):
+        """
+        Creates n+k colors, where n is length of VOI list and k is size of a buffer for VOIs added in runtime.
+        Should be called after ending series of add_voi calls.
+
+        :param int k: size of color buffer, default is 3
+        """
+        n = len(self.vois)
+        # create list of HSV tuples, where saturation and value in maxed, only hue is changing
+        # so there are n+k colors picked from HSV cone base circumference
+        hsv_tuples = [(x * 1.0 / (n + k), 1, 1) for x in range(n + k)]
+        # map HSV to RGB
+        rgb_tuples = list(map(lambda x: colorsys.hsv_to_rgb(*x), hsv_tuples))
```

Comment:
What about list comprehension ?

```
rgb_colors = [colorsys.hsv_to_rgb(*color_tuple) for color_tuple in hsv_tuples]
```

Reply:
Right, my functional programmer side was awakened while I was writing it

Reply:
Done